src/wallet/spend.cpp

```diff
@@ -573,6 +573,28 @@ util::Result<SelectionResult> ChooseSelectionResult(const CWallet& wallet, const
                              "Please try sending a smaller amount or manually consolidating your wallet's UTXOs")};
     }
 
+    // If the chosen input set has unconfirmed inputs, check for synergies from overlapping ancestry
+    for (auto& result : eligible_results) {
+        std::vector<COutPoint> outpoints;
+        CAmount summed_bump_fees = 0;
+        std::set<COutput> coins = result.GetInputSet();
+        for (auto& coin : coins) {
+            if (coin.depth > 0) continue; // Bump fees only exist for unconfirmed inputs
+            outpoints.push_back(coin.outpoint);
+            summed_bump_fees += coin.ancestor_bump_fees;
+        }
+        std::optional<CAmount> grouped_bump_fees = wallet.chain().CalculateTotalBumpFees(outpoints, coin_selection_params.m_effective_feerate);
+        if (!grouped_bump_fees.has_value()) {
+            return util::Error{_("Failed to calculate bump fees, because unconfirmed UTXOs depend on enormous cluster of unconfirmed transactions.")};
+        }
+        CAmount bump_fee_overestimate = summed_bump_fees - grouped_bump_fees.value();
+        if (bump_fee_overestimate) {
+            result.SetBumpFeeDiscount(bump_fee_overestimate);
+            // Update waste
+            result.ComputeAndSetWaste(coin_selection_params.min_viable_change, coin_selection_params.m_cost_of_change, coin_selection_params.m_change_fee);
```

Comment:
Still open

Reply:
Did about 40M executions, and no issues! Gonna calculate the Waste score below here as discussed.

Reply:
In theory the waste score should be the same.

Because not all of the params are available within BnB we call
`result.ComputeAndSetWaste(cost_of_change, cost_of_change, 0);`
instead of
`result.ComputeAndSetWaste(min_viable_change, cost_of_change, change_fee);`

I believe this should be fine as BnB will not produce results with more than `cost_of_change` excess and therefore `GetChange()` will always return 0 and then `change_fee` doesn't matter.